eclipse-theia/theia
# Support for terminal profiles.

#### What it does
Adds support for terminal profiles. 

Fixes [#11503](https://github.com/eclipse-theia/theia/issues/11503)

* UI and service to manage terminal profiles
* Handle profiles in preferences according to VS Code schema
* API and contribution markup for contributing profiles and activation event handling

contributed on behalf of STMicroelectronics

#### How to test
This PR adds two new items under the "Terminal" menu: "New Terminal...", which allows to select a profile and use it to open a terminal and "Choose Default Profile", which selects the profile to be used when selecting "New Terminal" from the menu.
Some profiles are added by default ("cmd" by default on windows), but they can also be contributed by plugins or defined in the preferences. 
Here are two extensions that can be used to test contributed terminals:

[plugins.zip](https://github.com/eclipse-theia/theia/files/10409008/plugins.zip)

One contributes a shell terminal profile ("JavaScript Debug Terminal") and the other a pseudo-pty terminal ("My Extension Terminal"). 

Adding profiles via terminals is described here:

https://code.visualstudio.com/docs/terminal/profiles

Note that terminal "source"s have not been implemented. Theia will accept the syntax, but the resulting profiles will not work.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
